unicore/hub/service/app_api.py

```diff
@@ -0,0 +1,107 @@
+from pyramid.exceptions import NotFound, Forbidden
+
+from cornice import Service
+from cornice.resource import resource, view
+
+from unicore.hub.service.models import App
+from unicore.hub.service.schema import App as AppSchema
+from unicore.hub.service import utils
+
+
+def require_principal(request, principals):
+    effective_principals = set(request.effective_principals)
+    accepted_principals = set(principals)
+    if not effective_principals.intersection(accepted_principals):
+        raise Forbidden
+
+
+def get_app_object(request):
+    app_id = request.matchdict['app_id']
+    app = request.db.query(App).get(app_id)
+
+    if app is None:
+        raise NotFound
+
+    return app
+
+
+@resource(collection_path='/apps', path='/apps/{app_id:\d+}')
```

Comment:
Our initial idea was that we'd use UUID for everything. So all our content types (User, Comments, Images, Polls, Pages) will be identified by a UUID